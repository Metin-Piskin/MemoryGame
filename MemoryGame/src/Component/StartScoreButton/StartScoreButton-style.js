import { StyleSheet } from "react-native";

export default StyleSheet.create({
    tablebutton: {
        position: 'absolute',
        right: 30,
        bottom: 40
    },
    table: {
        padding: 10,
        borderRadius: 10
    }
})