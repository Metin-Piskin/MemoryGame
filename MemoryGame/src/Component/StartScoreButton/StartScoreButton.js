import React from 'react';
import { TouchableOpacity } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5';

import styles from './StartScoreButton-style';

const StartScoreButton = ({ onPress }) => {
    return (
        <TouchableOpacity
            onPress={onPress}
            style={styles.tablebutton}
        >
            <LinearGradient
                style={styles.table}
                colors={[
                    '#1e293b',
                    '#1e293b',
                    '#334155',
                    '#334155',
                ]}
            >
                <FontAwesome5
                    name='table'
                    color={'#fff'}
                    size={30}
                />
            </LinearGradient>
        </TouchableOpacity>
    )
}

export default StartScoreButton;