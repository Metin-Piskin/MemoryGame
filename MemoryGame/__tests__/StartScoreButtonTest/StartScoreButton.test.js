import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

import StartScoreButton from '../../src/Component/StartScoreButton';
import StartScoreButtonStyles from '../../src/Component/StartScoreButton/StartScoreButton-style';

jest.mock('react-native-linear-gradient', () => 'LinearGradient')
jest.mock('react-native-vector-icons/FontAwesome5', () => 'FontAwesome5')

test('renders correctly', () => {
    const comp = render(<StartScoreButton />);
    expect(comp).toMatchSnapshot();
});

test('should trigerr onPress', () => {
    const mockFunction = jest.fn();
    const comp = render(<StartScoreButton onPress={mockFunction} />);

    const StartScoreButtonPress = comp.getByTestId('start-score-button-touchableopacity');
    fireEvent(StartScoreButtonPress, 'press');

    expect(mockFunction).toBeCalled();
})

test('should render default theme style', () => {
    const comp = render(<StartScoreButton />);
    const StartScoreButtonStyle = comp.getByTestId('start-score-button-touchableopacity').props.style;

    expect(StartScoreButtonStyle).toMatchObject(StartScoreButtonStyles.tablebutton);
});

test('should render default theme style', () => {
    const comp = render(<StartScoreButton />);
    const StartScoreButtonStyle = comp.getByTestId('start-score-button-lineargradient').props.style;

    expect(StartScoreButtonStyle).toMatchObject(StartScoreButtonStyles.table);
});  
