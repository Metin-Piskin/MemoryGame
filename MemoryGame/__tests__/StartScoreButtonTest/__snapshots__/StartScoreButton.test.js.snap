// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "bottom": 40,
      "opacity": 1,
      "position": "absolute",
      "right": 30,
    }
  }
  testID="start-score-button-touchableopacity"
>
  <LinearGradient
    colors={
      [
        "#1e293b",
        "#1e293b",
        "#334155",
        "#334155",
      ]
    }
    style={
      {
        "borderRadius": 10,
        "padding": 10,
      }
    }
    testID="start-score-button-lineargradient"
  >
    <FontAwesome5
      color="#fff"
      name="table"
      size={30}
    />
  </LinearGradient>
</View>
`;
